### ISSUE TITLE
NoCaps test results  much lower than reported val results

### ISSUE THREAD

**Issue by YovaKem** [salesforce/BLIP#87]
I ran inference on the test set of NoCaps using the code from the demo:

```
from models.blip import blip_decoder

image_size = 384
image = load_demo_image(image_size=image_size, device=device)

model_url = 'https://storage.googleapis.com/sfr-vision-language-research/BLIP/models/model_base_capfilt_large.pth'
    
model = blip_decoder(pretrained=model_url, image_size=image_size, vit='base')
model.eval()
model = model.to(device)

with torch.no_grad():
    # beam search
    caption = model.generate(image, sample=False, num_beams=3, max_length=20, min_length=5) 
    # nucleus sampling
    # caption = model.generate(image, sample=True, top_p=0.9, max_length=20, min_length=5) 
    print('caption: '+caption[0])
```

I tried both beam search and nucleus sampling and the results with both are much lower than what is reported in the paper. Briefly, the paper reports CIDEr 109.6 on the entire NoCaps dataset with BLIP_capfilt_large and I get 74.29 with beam search and 45.3 with nucleus sampling. I realise that one is measured on the validation data and the other on test data, but this seems like a huge difference. So I wonder if you can point me to what changes I need to make in the above code to get the correct results.

Thanks!

**Comment by YovaKem** [comment]
I just noticed [this](https://github.com/salesforce/BLIP/blob/main/eval_nocaps.py) script so I'm closing this.
